js/reference.js

```diff
@@ -496,8 +509,7 @@ var ERMrest = (function(module) {
      * @returns {ERMrest.Reference} The copy of the reference object.
      */
     function _referenceCopy(source) {
-        // TODO: make a (copy-on-write) copy of the source reference
-        return source; // TODO
+        return Object.assign(Object.create(Reference.prototype), source);
```

Comment:
@jrchudy 

This method of creating objects without invoking their constructor is really good. 

Should we make **copy** part of some utility function, instead of declaring it in this class ? So we can use it in other classes too.

For example
```javascript
module._copy = function(source, destClass) {
    var obj = {};
   
    // destClass parameter is optional, if you just want to clone an existing object
    if (destClass) obj = Object.create(destClass.prototype);
   
    return Object.assign(obj, source);
}; 
```